app/models/assignment.rb

```diff
@@ -37,26 +35,14 @@ def for_presentation(built_at: nil)
     private
 
     def presentation_query
-      joins(:split).select(select_sql)
-    end
-
-    def select_sql
-      @select_sql ||= (passthrough_sql_columns + overridden_sql_columns).join(",\n")
-    end
-
-    def passthrough_sql_columns
-      passthrough_columns.map { |n| "assignments.#{connection.quote_column_name(n)}" }
-    end
-
-    def passthrough_columns
-      (column_names - NON_PRESENTATION_COLUMNS - %w(variant))
+      Assignment.from(joins(:split).select(select_columns), :assignments).joins(:split).readonly
```

Comment:
This (making the assignments query into a subselect) turned out to be important so that ActiveRecord doesn't assume that it can go straight to the variant column as a source of truth when doing joins. It was wiping out my calculated variant value.

I also refactored for simplicity a little bit in the process.

Reply:
(this went away)